### ISSUE TITLE
improve stack frame generation

### ISSUE THREAD

**Issue by LADSoft** [LADSoft/OrangeC#591]
clang modifies the push/pop model for function call stack frames to instead allocate the maximum stack frame needed for all function calls within a function body, then never do push/pops at least not for __cdecl style functions.    This would also be a much cleaner way of handling the stackalign command line parameter and also various stack-based temporaries the backend creates by pushing something on the stack.

**Comment by GitMensch** [comment]
This sounds like a low hanging fruit... Maybe target that before the mega-milestone "optimization"?

**Comment by LADSoft** [comment]
yeah this should  be easy enough...   I should do this and #709 at the same time I guess...
